common/unsafe/src/main/java/org/apache/spark/unsafe/array/ByteArrayMethods.java

```diff
@@ -48,6 +49,16 @@ public static int roundNumberOfBytesToNearestWord(int numBytes) {
   public static int MAX_ROUNDED_ARRAY_LENGTH = Integer.MAX_VALUE - 15;
 
   private static final boolean unaligned = Platform.unaligned();
+  /**
+   * MemoryBlock equality check for MemoryBlocks.
+   * @return true if the arrays are equal, false otherwise
+   */
+  public static boolean arrayEqualsBlock(
```

Comment:
no one is calling it.

Reply:
Thank you for your review. I will address them on Wed or Thu.

Reply:
Thank you for your comments. I will address them on Wed or Thu.

Reply:
This method is called from `UTF8String` class.

Reply:
I think this only works for `ByteArrayMemoryBlock`, doesn't it?